src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2373,6 +2376,10 @@ protected virtual void ProcessMethod (MethodDefinition method, in DependencyInfo
 			if (ShouldParseMethodBody (method))
 				MarkMethodBody (method.Body);
 
+			if (method.DeclaringType.IsMulticastDelegate () && (method.Name == "BeginInvoke" || method.Name == "EndInvoke")) {
+				MarkMulticastDelegate (method.DeclaringType, markAllMethods: true);
```

Comment:
I don't understand the rationale behind this logic. Why not mark only BeginInvoke/EndInvoke pair when one of them is marked

Reply:
Just lest code to maintain - the net effect is the same. I can redo it to make it mark the other method instead.

Reply:
I'd prefer clarity over less code to maintain